python/servo/package_commands.py

```diff
@@ -179,6 +188,50 @@ def package(self, build_type: BuildType, flavor=None, with_asan=False):
             except subprocess.CalledProcessError as e:
                 print("Packaging Android exited with return value %d" % e.returncode)
                 return e.returncode
+        elif self.is_openharmony():
+            # hvigor doesn't support an option to place output files in a specific directory
+            # so copy the source files into the target/openharmony directory first.
+            ohos_app_dir = path.join(self.get_top_dir(), "support", "openharmony")
+            ohos_target_dir = path.join(
```

Comment:
The source files of OH app, shouldn't be different for release / debug mode, so I don't think adding `build_type.directory_name()` is strictly necessary.

Reply:
The idea is to allow the HAP (which will be generated inside the source folder hierarchy) for debug/release/production profiles to co-exist like we do for other targets.

Reply:
Hmm, does the output folder in hvigor not differ between release / debug profiles? (To be fair I've only ever built with the default profile, which I think is debug)

Reply:
I'm not sure either. The output currently ends up inside `<top-level for app source>/entry/build/default/outputs/default` . Is one of the `default` in this path referring to the debug/release type? There is a [buildMode](https://github.com/servo/servo/blob/a5798ceadae9c3834db3afa03cbb477452128e73/support/openharmony/build-profile.json5#L20) declaration, but I don't follow how that translates to "default". Also, shouldn't the invocation of the `hvigor` differ for debug/release? It is currently the same for both.

Reply:
Just tried invoking hvigor with `-p buildMode=release` and while the build successfully completes successfully, it does give this warning
```
> hvigor Finished :servoshell:default@CompileResource... after 53 ms 
> hvigor WARN: Dependency libservoshell.so not found.

```
but the HAP does contain the `libservoshell.so`. The path is still the same as without `buildMode=release` i.e  `entry/build/default/outputs/default` so I guess the path doesn't seem to be affected by the buildMode.

Reply:
@jschwe the "dependency not found" error seems to originate from 'obfustcation-config-resolver.js' in the ohos-plugin module. I don't know why it says the .so is not found, but still the HAP has the library and it runs as expected. 

I've added code in [a797e9a](https://github.com/servo/servo/pull/33295/commits/a797e9a3b1d3782b269632e1731e5a1487e57ec3) . I've not removed the profile subfolder under `target/openharmony/<triple>` since the output path doesn't seem to change.